### ISSUE TITLE
eslint-fix is not working

### ISSUE THREAD

**Issue by elvis-sik** [aaronjensen/spacemacs.d#3]
Hi @aaronjensen 

If you have the time, could you help me try to make your layer work in my machine?

`eslint-fix` gives the following error: `zsh:1: command not found: --stdin`. As expected by skimming the source for that function, the problem is the variable `flycheck-javascript-eslint-executable`, which I checked and indeed is `nil` when this error occurs. So the function ends up with `--stdin` as the first thing in the command call.

Now, I don't know enough about `flycheck` internals to make sense of this, but flycheck **is** working, and is pointing to eslint-detected errors.

Again, no problem if you cannot take the time to help me set things up here.

**Comment by aaronjensen** [comment]
Howdy, sure. I definitely need to add some documentation and resiliency to this if other people are going to use it. I may release it as a package one day, but I haven't had a chance yet.

So there are likely two things that will make this not work. You are right, if `flycheck-javascript-eslint-executable` is not set, it will not work. More than that, however, is that this relies on [eslint_d](https://github.com/mantoni/eslint_d.js) rather than regular `eslint` for its performance and `--fix-to-stdout` option, which `eslint` does not yet have. All of this results in a much faster linting/fixing than could be done w/ `eslint` alone.

In my javascript layer there are a few moving parts that make it all work. 

1. The [add-node-modules-path](https://github.com/codesuki/add-node-modules-path) package.
2. [Hook](https://github.com/aaronjensen/spacemacs.d/blob/0989464177e5c5514d5ad303812f7c214a42890d/layers/%2Baj/aj-javascript/packages.el#L33) for [aj-javascript/eslintd-set-flycheck-executable](https://github.com/aaronjensen/spacemacs.d/blob/0989464177e5c5514d5ad303812f7c214a42890d/layers/%2Baj/aj-javascript/funcs.el#L1-L5)
3. [Hook](https://github.com/aaronjensen/spacemacs.d/blob/0989464177e5c5514d5ad303812f7c214a42890d/layers/%2Baj/aj-javascript/packages.el#L15-L16) for `eslint-fix`

Hopefully this helps you get it set up, if not, feel free to ask more questions.

**Comment by aaronjensen** [comment]
Also, this is a much simpler (and probably slower) version that is already on melpa: https://github.com/codesuki/eslint-fix

**Comment by aaronjensen** [comment]
I've renamed the package to eslintd-fix and improved on it some. It is now a minor mode with its own customization and default: https://github.com/aaronjensen/spacemacs.d/blob/master/layers/+aj/aj-javascript/local/eslintd-fix/eslintd-fix.el

To use it, [add a hook like so](https://github.com/aaronjensen/spacemacs.d/blob/c7513b90f44ef0df11a20ec608b381c4d1d57e06/layers/%2Baj/aj-javascript/packages.el#L15).

As long as `eslint_d` is in your path, it should "just work". 

I'll keep working on it and eventually release it on melpa. If you want to give it a shot and let me know if it works/doesn't I'd appreciate it. Thanks!

**Comment by elvis-sik** [comment]
@aaronjensen thanks for being so helpful 😄 

I pulled your modifications and installed `eslint_d`. I ran a quick test and it seems to be working.

You plan to release `eslintd-fix` to melpa right? I think this functionality would be good to have in Spacemacs.

If I run into any issues I'll report. I am happy that saving fixes those small linting issues now, I already had that in Python and man, is it good.
